sample/sample_test.go

```diff
@@ -1,37 +1,124 @@
 
 import (
+	"io"
 	"os"
 	"path/filepath"
 	"testing"
+	"time"
+
+	"github.com/cuducos/minha-receita/download"
 )
 
 var testdata = filepath.Join("..", "testdata")
 
-func TestSample(t *testing.T) {
+func testdataWithoutUpdatedAt(t *testing.T) string {
 	tmp := t.TempDir()
-	if err := Sample(testdata, tmp, 42); err != nil {
-		t.Fatalf("expected no error running sample, got %s", err)
-	}
-	ls, err := os.ReadDir(tmp)
+	ls, err := filepath.Glob(filepath.Join(testdata, "*"))
 	if err != nil {
-		t.Errorf("expected no error readind dir %s, got %s", tmp, err)
+		t.Fatalf("could not read %s", testdata)
 	}
-
-	expected := 12
-	var got []string
 	for _, f := range ls {
-		if !f.IsDir() {
-			got = append(got, f.Name())
+		if filepath.Base(f) == download.FederalRevenueUpdatedAt {
+			continue
 		}
-	}
-	if len(got) != expected {
-		t.Errorf("expected %d files in the sample directory, got %d: %v", expected, len(got), got)
-	}
-	for _, f := range got {
-		_, err := os.Stat(filepath.Join(testdata, f))
+		s, err := os.Stat(f)
 		if err != nil {
-			t.Errorf("expected %s to exist in %s, but it does not", f, testdata)
+			t.Fatalf("could not read %s", f)
+		}
+		if s.IsDir() || !s.Mode().IsRegular() {
+			continue
 		}
+		func() {
+			r, err := os.Open(f)
+			if err != nil {
+				t.Fatalf("could not open %s", f)
+			}
+			defer r.Close()
+			d := filepath.Join(tmp, filepath.Base(f))
+			w, err := os.Create(d)
+			if err != nil {
+				t.Fatalf("could not create %s", d)
+			}
+			defer w.Close()
+			_, err = io.Copy(w, r)
+			if err != nil {
+				t.Fatalf("could not copy %s to %s", f, d)
+			}
+		}()
+	}
+	return tmp
+}
+
+var defaultUpdatedAt = time.Now().Format("2006-01-02")
+
+func TestSample(t *testing.T) {
+	testCases := []struct {
+		desc            string
+		srcDir          string
+		outDir          string
+		updatedAt       string
+		expectedSamples int
+		err             bool
+	}{
+		{
+			"Copy existing updated_at.txt file",
+			testdata,
+			t.TempDir(),
+			defaultUpdatedAt,
+			13,
+			false,
+		},
+		{
+			"Ignore updated_at.txt file",
+			testdataWithoutUpdatedAt(t),
+			t.TempDir(),
+			"",
+			12,
+			false,
+		},
+		{
+			"Mock updated_at.txt file with invalid date format",
+			testdataWithoutUpdatedAt(t),
+			t.TempDir(),
+			"17-10-2022",
+			12,
+			true,
+		},
+		{
+			"Mock updated_at.txt file",
+			testdataWithoutUpdatedAt(t),
+			t.TempDir(),
+			defaultUpdatedAt,
+			13,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			if err := Sample(tc.srcDir, tc.outDir, 42, tc.updatedAt); err != nil {
+				if tc.err {
+					return
+				}
+
+				t.Fatalf("expected no error running sample, got %s", err)
+			}
+
+			ls, err := os.ReadDir(tc.outDir)
+			if err != nil {
+				t.Errorf("expected no error reading dir %s, got %s", tc.outDir, err)
+			}
+
+			var got []string
+			for _, f := range ls {
+				if !f.IsDir() {
+					got = append(got, f.Name())
+				}
+			}
+			if len(got) != tc.expectedSamples {
```

Comment:
A gente não precisa de um `[]string` se vamos apenas contar os arquivos. Qua tal simplificar?

```suggestion
			var got int
			for _, f := range ls {
				if !f.IsDir() {
					got += 1
				}
			}
			if got != tc.expectedSamples {
```